coffee-jpa/src/main/java/hu/icellmobilsoft/coffee/jpa/helper/TransactionHelper.java

```diff
@@ -0,0 +1,301 @@
+import hu.icellmobilsoft.coffee.tool.common.FunctionalInterfaces.BaseExceptionFunction6;
+import hu.icellmobilsoft.coffee.tool.common.FunctionalInterfaces.BaseExceptionRunner;
+import hu.icellmobilsoft.coffee.tool.common.FunctionalInterfaces.BaseExceptionSupplier;
+
+/**
+ * Helper class for managing transactions. <br>
+ * This class is an alternative solution based on {@link org.apache.deltaspike.jpa.api.transaction.TransactionHelper}.
+ *
+ * @author csaba.balogh
+ * @since 1.12.0
+ */
+@Dependent
+public class TransactionHelper {
+
+    private static final String AT_LEAST_ONE_INCOMING_PARAMETER_IS_NULL_OR_BLANK = "At least one incoming parameter is NULL or BLANK!";
+
+    /**
+     * Execute the given {@link BaseExceptionSupplier} inside a {@link Transactional} block.
+     *
+     * @param baseExceptionSupplier
+     *            the given {@link BaseExceptionSupplier}.
+     * @return the return value of the executed {@link BaseExceptionSupplier}.
+     * @param <R>
+     *            the type of the return value.
+     * @throws BaseException
+     *             in case of any exception occurs during the process.
+     */
+    @Transactional
+    public <R> R executeWithTransaction(BaseExceptionSupplier<R> baseExceptionSupplier) throws BaseException {
+        if (Objects.isNull(baseExceptionSupplier)) {
+            throw newInvalidParameterException("baseExceptionSupplier is NULL!");
+        }
+        return baseExceptionSupplier.get();
+    }
+
+    /**
+     * Execute the given {@link BaseExceptionRunner} inside a {@link Transactional} block.
+     * 
+     * @param baseExceptionRunner
+     *            the given {@link BaseExceptionRunner}.
+     * @throws BaseException
+     *             in case of any exception occurs during the process.
+     */
+    @Transactional
+    public void executeWithTransaction(BaseExceptionRunner baseExceptionRunner) throws BaseException {
+        if (Objects.isNull(baseExceptionRunner)) {
+            throw newInvalidParameterException("baseExceptionRunner is NULL!");
+        }
+        baseExceptionRunner.run();
+    }
+
+    /**
+     * Execute the given {@link BaseExceptionFunction} inside a {@link Transactional} block.
+     * 
+     * @param baseExceptionFunction
+     *            the given {@link BaseExceptionFunction}.
+     * @param p1
+     *            the first parameter.
+     * @return the return value of the executed {@link BaseExceptionFunction}.
+     * @param <P1>
+     *            the type of the first parameter.
+     * @param <R>
+     *            the type of the return value.
+     * @throws BaseException
+     *             in case of any exception occurs during the process.
+     */
+    @Transactional
+    public <P1, R> R executeWithTransaction(BaseExceptionFunction<P1, R> baseExceptionFunction, P1 p1) throws BaseException {
+        if (isNullOrBlankAnyParameter(baseExceptionFunction, p1)) {
```

Comment:
nem feltetlenul kell kenyszeriteni a not null parametereket. Ezek altalanos metodusok, barmilyen parameter kombinacioval lehetnek hasznalva.